mitodl/mitxonline
# Upgrade dialog choose any course run

### What are the relevant tickets?
Fix https://github.com/mitodl/hq/issues/3671 and partially https://github.com/mitodl/hq/issues/3109

### Description (What does it do?)
Changes the enroll now button to open the dialog whenever there is more then one course run available for enrollment.
In the dialog the dropdown select course run shows all enrollable course runs. The runs that can't be upgraded have a label  `(no certificate available)`




### How can this be tested?
There are few scenarios to test:
**1. All course runs for a given course don't have products but are enrollable**
<img width="746" alt="Screenshot 2024-03-20 at 2 19 36 PM" src="https://github.com/mitodl/mitxonline/assets/7574259/3a749416-5a21-4d25-a569-790c309dc9d4">

**2. One course run has product and other not, but both are enrollable**
<img width="747" alt="Screenshot 2024-03-20 at 2 33 10 PM" src="https://github.com/mitodl/mitxonline/assets/7574259/ea377894-bdcb-434d-a649-935d268e4758">

**3. two course runs both have products both are enrollable, but only one is upgradable** (meaning payment_deadline is in the future) - in this scenario the dialog looks the same as case 2.
**4. Has one enrollable course run with no product** - in this case when you click enroll button it just enrolls you, no dialog is shown.
**5. Has one enrollable and upgradable run**   
<img width="747" alt="Screenshot 2024-03-20 at 2 35 41 PM" src="https://github.com/mitodl/mitxonline/assets/7574259/5de5e4c2-292f-4978-b391-12bb2b872dec">

**6. has two runs both self paced and enrollable** ( and upgradable?)
If at least one of the course runs is upgradable the dialog would look like case 2. Otherwise, the dialog would show up as case 1.
